ui/pages/confirmations/components/confirm/info/approve/approve-static-simulation/approve-static-simulation.tsx

```diff
@@ -0,0 +1,106 @@
+import { NameType } from '@metamask/name-controller';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { useEffect, useState } from 'react';
+import Name from '../../../../../../../components/app/name';
+import { Box, Text } from '../../../../../../../components/component-library';
+import Tooltip from '../../../../../../../components/ui/tooltip';
+import {
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+  BorderRadius,
+  Display,
+  TextAlign,
+} from '../../../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { useConfirmContext } from '../../../../../context/confirm';
+import { useAssetDetails } from '../../../../../hooks/useAssetDetails';
+import StaticSimulation from '../../shared/static-simulation/static-simulation';
+import { Container } from '../../shared/transaction-data/transaction-data';
+import {
+  UNLIMITED_MSG,
+  useApproveTokenSimulation,
+} from '../hooks/use-approve-token-simulation';
+import { useIsNFT } from '../hooks/use-is-nft';
+
+export const ApproveStaticSimulation = () => {
+  const t = useI18nContext();
+
+  const { currentConfirmation: transactionMeta } = useConfirmContext() as {
+    currentConfirmation: TransactionMeta;
+  };
+
+  const { decimals: initialDecimals } = useAssetDetails(
+    transactionMeta?.txParams?.to,
+    transactionMeta?.txParams?.from,
+    transactionMeta?.txParams?.data,
+  );
+
+  // decimals are sometimes undefined for a split second, so we need to ensure
+  // the spending cap is recalculated when the value updates
+  const [decimals, setDecimals] = useState(initialDecimals || '0');
+
+  useEffect(() => {
```

Comment:
I'm not sure what purpose this `useEffect` serves and why we need a `useState`?

Could we just set decimals as a standard variable based on `initialDecimals`?